src/libstore/realisation.hh

```diff
@@ -28,8 +39,60 @@ struct Realisation {
 
     nlohmann::json toJSON() const;
     static Realisation fromJSON(const nlohmann::json& json, const std::string& whence);
+
+    StorePath getPath() const { return outPath; }
+
+    GENERATE_CMP(Realisation, me->id, me->outPath);
+};
+
+struct OpaquePath {
+    StorePath path;
+
+    StorePath getPath() const { return path; }
+
+    GENERATE_CMP(OpaquePath, me->path);
 };
 
-typedef std::map<DrvOutput, Realisation> DrvOutputs;
+
+/**
+ * A store path with all the history of how it went into the store
+ */
+struct RealisedPath {
```

Comment:
Is all this C++ worth it? In derivation.hh we just wrapped a variant and called it a day. Other places we `typedef`ed the variant. It would be good to be consistent whatever route we choose, I suppose.

Reply:
Yeah I eventually removed because it was a lot of boilerplate for nearly nothing